### PR TITLE
update contributing guide

### PR DESCRIPTION
**Pull request by MarkusZoppelt**
### Motivation & Context
The guide in CONTRIBUTING.md was outdated and did not include prerequisites for Bazel and Java (Maven / Gradle).

### Description

- docs(README): fix order of instructions and add a comment for bazelisk
- docs(CONTRIBUTING): update contributing guide with Bazel and Java prerequisites

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,23 +8,39 @@ For more information, see [how to create and publish a new release](./docs/RELEA
...
 * [LLVM >= 14](https://clang.llvm.org/get_started.html)
 * [go >= 1.18](https://go.dev/doc/install)
 * [libcap](https://man7.org/linux/man-pages/man3/libcap.3.html)
 
+Depending on what build system / programming language you want to use with
+cifuzz, you might also need these prerequisites:
+
+* [make](https://www.gnu.org/software/make/)
+* [CMake >= 3.21](https://cmake.org/)
+* [Bazel >= 5.3.1](https://bazel.build/install)
```
Comment (fmeum) [inline CONTRIBUTING.md:20]: Let's make this 5.3.2 instead, 5.3.1 had a bug.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,23 +8,39 @@ For more information, see [how to create and publish a new release](./docs/RELEA
 
 ### Prerequisites
 * [git](https://git-scm.com/)
-* [make](https://www.gnu.org/software/make/)
-* [CMake >= 3.21](https://cmake.org/)
 * [LLVM >= 14](https://clang.llvm.org/get_started.html)
 * [go >= 1.18](https://go.dev/doc/install)
 * [libcap](https://man7.org/linux/man-pages/man3/libcap.3.html)
 
+Depending on what build system / programming language you want to use with
```
Comment (fmeum) [inline CONTRIBUTING.md:15]: This sounds targeted towards users rather than devs. We should mention that these are required to run the tests.

**Review by fmeum — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (MarkusZoppelt) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 2 above)
Comment (MarkusZoppelt) [reply CONTRIBUTING.md]: I divided the list in "build dependencies" and "test dependencies"

**Pull request merged**